--- net/minecraft/client/network/NetHandlerPlayClient.java
+++ net/minecraft/client/network/NetHandlerPlayClient.java
@@ -763,6 +763,10 @@
 		} else {
 			this.client.ingameGUI.getChatGUI().printChatMessage(packetIn.getChatComponent());
 		}
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleChat(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
@@ -1125,6 +1129,10 @@
 	public void handleBlockAction(SPacketBlockAction packetIn) {
 		PacketThreadUtil.checkThreadAndEnqueue(packetIn, this, this.client);
 		this.client.world.addBlockEvent(packetIn.getBlockPosition(), packetIn.getBlockType(), packetIn.getData1(), packetIn.getData2());
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleBlockAction(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
@@ -1194,6 +1202,10 @@
 		MapData mapdata = ItemMap.loadMapData(packetIn.getMapId(), this.client.world);
 		packetIn.setMapdataTo(mapdata);
 		this.client.entityRenderer.getMapItemRenderer().updateMapTexture(mapdata);
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleMaps(this, packetIn);
+		/* <<< WDL */
 	}
 
 	public void handleEffect(SPacketEffect packetIn) {
@@ -1562,6 +1574,10 @@
 			String s1 = packetbuffer2.readString(256);
 			this.client.getSoundHandler().stop(s1, SoundCategory.getByName(s));
 		}
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleCustomPayload(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
