--- net/minecraft/client/network/NetHandlerPlayClient.java
+++ net/minecraft/client/network/NetHandlerPlayClient.java
@@ -257,7 +257,7 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class NetHandlerPlayClient implements INetHandlerPlayClient {
+public class NetHandlerPlayClient implements INetHandlerPlayClient/* WDL >>> */, wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
 	private static final Logger LOGGER = LogManager.getLogger();
 	private final NetworkManager netManager;
 	private final GameProfile profile;
@@ -704,6 +704,10 @@
 	}
 
 	public void processChunkUnload(SPacketUnloadChunk packetIn) {
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleChunkUnload(this, this.world, packetIn);
+		/* <<< WDL */
+
 		PacketThreadUtil.checkThreadAndEnqueue(packetIn, this, this.client);
 		this.world.doPreChunk(packetIn.getX(), packetIn.getZ(), false);
 	}
@@ -727,6 +731,10 @@
 	 * Invoked when disconnecting, the parameter is a ChatComponent describing the reason for termination
 	 */
 	public void onDisconnect(ITextComponent reason) {
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCDisconnect(this, reason);
+		/* <<< WDL */
+
 		this.client.loadWorld((WorldClient)null);
 
 		if (this.guiScreenServer != null) {
@@ -775,6 +783,10 @@
 	public void handleChat(SPacketChat packetIn) {
 		PacketThreadUtil.checkThreadAndEnqueue(packetIn, this, this.client);
 		this.client.ingameGUI.addChatMessage(packetIn.getType(), packetIn.getChatComponent());
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleChat(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
@@ -1146,6 +1158,10 @@
 	public void handleBlockAction(SPacketBlockAction packetIn) {
 		PacketThreadUtil.checkThreadAndEnqueue(packetIn, this, this.client);
 		this.client.world.addBlockEvent(packetIn.getBlockPosition(), packetIn.getBlockType(), packetIn.getData1(), packetIn.getData2());
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleBlockAction(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
@@ -1234,6 +1250,10 @@
 
 		packetIn.setMapdataTo(mapdata);
 		mapitemrenderer.updateMapTexture(mapdata);
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleMaps(this, packetIn);
+		/* <<< WDL */
 	}
 
 	public void handleEffect(SPacketEffect packetIn) {
@@ -1656,6 +1676,10 @@
 			String s1 = packetbuffer3.readString(256);
 			this.client.getSoundHandler().func_189520_a(s1, SoundCategory.func_187950_a(s));
 		}
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleCustomPayload(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
