--- net/minecraft/client/multiplayer/WorldClient.java
+++ net/minecraft/client/multiplayer/WorldClient.java
@@ -43,7 +43,7 @@
 import net.minecraft.world.storage.SaveHandlerMP;
 import net.minecraft.world.storage.WorldInfo;
 
-public class WorldClient extends World {
+public class WorldClient extends World/* WDL >>> */ implements wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
 	private NetHandlerPlayClient connection;
 	private ChunkProviderClient field_73033_b;
 	private final Set<Entity> field_73032_d = Sets.<Entity>newHashSet();
@@ -94,6 +94,10 @@
 		this.profiler.endStartSection("blocks");
 		this.func_147456_g();
 		this.profiler.endSection();
+
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientTick(this);
+		/* <<< WDL */
 	}
 
 	public void func_73031_a(int p_73031_1_, int p_73031_2_, int p_73031_3_, int p_73031_4_, int p_73031_5_, int p_73031_6_) {
@@ -232,6 +236,10 @@
 	}
 
 	public Entity removeEntityFromWorld(int p_73028_1_) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, p_73028_1_);
+		/* <<< WDL */
+
 		Entity entity = (Entity)this.entitiesById.remove(p_73028_1_);
 
 		if (entity != null) {
