--- net/minecraft/client/gui/GuiIngameMenu.java
+++ net/minecraft/client/gui/GuiIngameMenu.java
@@ -7,7 +7,7 @@
 import net.minecraft.client.resources.I18n;
 import net.minecraft.realms.RealmsBridge;
 
-public class GuiIngameMenu extends GuiScreen {
+public class GuiIngameMenu extends GuiScreen/* WDL >>> */ implements wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
 	private int field_146445_a;
 	private int field_146444_f;
 
@@ -29,9 +29,17 @@
 		this.buttons.add(new GuiButton(5, this.width / 2 - 100, this.height / 4 + 48 + i, 98, 20, I18n.format("gui.achievements", new Object[0])));
 		this.buttons.add(new GuiButton(6, this.width / 2 + 2, this.height / 4 + 48 + i, 98, 20, I18n.format("gui.stats", new Object[0])));
 		guibutton.active = this.minecraft.isSingleplayer() && !this.minecraft.getIntegratedServer().getPublic();
+
+		/* WDL >>> */
+		wdl.WDLHooks.injectWDLButtons(this, buttons, buttons::add);
+		/* <<< WDL */
 	}
 
 	protected void actionPerformed(GuiButton p_146284_1_) throws IOException {
+		/* WDL >>> */
+		wdl.WDLHooks.handleWDLButtonClick(this, p_146284_1_);
+		/* <<< WDL */
+
 		switch (p_146284_1_.id) {
 		case 0:
 			this.minecraft.displayGuiScreen(new GuiOptions(this, this.minecraft.gameSettings));
