--- net/minecraft/inventory/InventoryBasic.java
+++ net/minecraft/inventory/InventoryBasic.java
@@ -9,13 +9,23 @@
 import net.minecraft.util.text.TextComponentString;
 import net.minecraft.util.text.TextComponentTranslation;
 
-public class InventoryBasic implements IInventory {
+public class InventoryBasic implements IInventory/* WDL >>> */, wdl.ducks.INetworkNameable /* <<< WDL */ {
 	private String inventoryTitle;
 	private int slotsCount;
 	private ItemStack[] inventoryContents;
 	private List<IInventoryChangedListener> listeners;
 	private boolean hasCustomName;
+	/* WDL >>> */
+	@Nullable
+	private String networkCustomName;
 
+	@Nullable
+	@Override
+	public String getCustomDisplayName() {
+		return networkCustomName;
+	}
+	/* <<< WDL */
+
 	public InventoryBasic(String p_i1561_1_, boolean p_i1561_2_, int p_i1561_3_) {
 		this.inventoryTitle = p_i1561_1_;
 		this.hasCustomName = p_i1561_2_;
@@ -25,6 +35,11 @@
 
 	public InventoryBasic(ITextComponent p_i45902_1_, int p_i45902_2_) {
 		this(p_i45902_1_.getString(), true, p_i45902_2_);
+		/* WDL >>> */
+		if (p_i45902_1_ instanceof TextComponentString) {
+			this.networkCustomName = p_i45902_1_.getString();
+		}
+		/* <<< WDL */
 	}
 
 	/**
