--- net/minecraft/crash/CrashReport.java
+++ net/minecraft/crash/CrashReport.java
@@ -17,7 +17,7 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class CrashReport {
+public class CrashReport/* WDL >>> */ implements wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
 	private static final Logger LOGGER = LogManager.getLogger();
 	private final String description;
 	private final Throwable cause;
@@ -95,6 +95,22 @@
 				return IntCache.func_85144_b();
 			}
 		});
+
+		/* WDL >>> */
+		try {
+			wdl.WDLHooks.onCrashReportPopulateEnvironment(this);
+		} catch (Throwable t) {
+			try {
+				final Logger LOGGER = LogManager.getLogger();
+				LOGGER.fatal("World Downloader: Failed to add crash info", t);
+				this.getCategory().addCrashSectionThrowable("World Downloader - Fatal error in crash handler (see log)", t);
+			} catch (Throwable t2) {
+				System.err.println("WDL: Double failure adding info to crash report!");
+				t.printStackTrace();
+				t2.printStackTrace();
+			}
+		}
+		/* <<< WDL */
 	}
 
 	/**
