--- net/minecraft/client/network/NetHandlerPlayClient.java
+++ net/minecraft/client/network/NetHandlerPlayClient.java
@@ -769,6 +769,10 @@
 		} else {
 			this.client.ingameGUI.getChatGUI().printChatMessage(packetIn.getChatComponent());
 		}
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleChat(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
@@ -1132,6 +1136,10 @@
 	public void handleBlockAction(SPacketBlockAction packetIn) {
 		PacketThreadUtil.checkThreadAndEnqueue(packetIn, this, this.client);
 		this.client.world.addBlockEvent(packetIn.getBlockPosition(), packetIn.getBlockType(), packetIn.getData1(), packetIn.getData2());
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleBlockAction(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
@@ -1201,6 +1209,10 @@
 		MapData mapdata = ItemMap.loadMapData(packetIn.getMapId(), this.client.world);
 		packetIn.setMapdataTo(mapdata);
 		this.client.entityRenderer.getMapItemRenderer().updateMapTexture(mapdata);
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleMaps(this, packetIn);
+		/* <<< WDL */
 	}
 
 	public void handleEffect(SPacketEffect packetIn) {
@@ -1569,6 +1581,10 @@
 			String s1 = packetbuffer2.readString(256);
 			this.client.getSoundHandler().func_189520_a(s1, SoundCategory.func_187950_a(s));
 		}
+
+		/* WDL >>> */
+		wdl.WDLHooks.onNHPCHandleCustomPayload(this, packetIn);
+		/* <<< WDL */
 	}
 
 	/**
