--- net/minecraft/client/gui/GuiIngameMenu.java
+++ net/minecraft/client/gui/GuiIngameMenu.java
@@ -32,12 +32,20 @@
 		guibutton.enabled = this.mc.isSingleplayer() && !this.mc.getIntegratedServer().getPublic();
 		this.buttonList.add(new GuiButton(5, this.width / 2 - 100, this.height / 4 + 48 + -16, 98, 20, I18n.format("gui.achievements", new Object[0])));
 		this.buttonList.add(new GuiButton(6, this.width / 2 + 2, this.height / 4 + 48 + -16, 98, 20, I18n.format("gui.stats", new Object[0])));
+
+		/* WDL >>> */
+		wdl.WDLHooks.injectWDLButtons(this, buttonList);
+		/* <<< WDL */
 	}
 
 	/**
 	 * Called by the controls from the buttonList when activated. (Mouse pressed for buttons)
 	 */
 	protected void actionPerformed(GuiButton button) throws IOException {
+		/* WDL >>> */
+		wdl.WDLHooks.handleWDLButtonClick(this, button);
+		/* <<< WDL */
+
 		switch (button.id) {
 		case 0:
 			this.mc.displayGuiScreen(new GuiOptions(this, this.mc.gameSettings));
