--- net/minecraft/client/resources/DefaultResourcePack.java
+++ net/minecraft/client/resources/DefaultResourcePack.java
@@ -15,7 +15,7 @@
 import net.minecraft.util.ResourceLocation;
 
 public class DefaultResourcePack implements IResourcePack {
-	public static final Set<String> field_110608_a = ImmutableSet.<String>of("minecraft", "realms");
+	public static final Set<String> field_110608_a = ImmutableSet.<String>of("minecraft", "realms"/* WDL >>> */, "wdl" /* <<< WDL */);
 	private final ResourceIndex field_188549_b;
 
 	public DefaultResourcePack(ResourceIndex p_i46541_1_) {
