--- net/minecraft/client/world/ClientWorld.java
+++ net/minecraft/client/world/ClientWorld.java
@@ -63,7 +63,7 @@
 import net.minecraft.world.storage.MapData;
 import net.minecraft.world.storage.WorldInfo;
 
-public class ClientWorld extends World {
+public class ClientWorld extends World/* WDL >>> */ implements wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
    private final List<Entity> globalEntities = Lists.newArrayList();
    private final Int2ObjectMap<Entity> entitiesById = new Int2ObjectOpenHashMap<>();
    private final ClientPlayNetHandler connection;
@@ -95,6 +95,10 @@
       this.chunkProvider.tick(hasTimeLeft);
       this.playMoodSoundAndCheckLight();
       this.getProfiler().endSection();
+
+      /* WDL >>> */
+      wdl.WDLHooks.onWorldClientTick(this);
+      /* <<< WDL */
    }
 
    public Iterable<Entity> getAllEntities() {
@@ -270,6 +274,10 @@
    }
 
    public void removeEntityFromWorld(int eid) {
+      /* WDL >>> */
+      wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, eid);
+      /* <<< WDL */
+
       Entity entity = this.entitiesById.remove(eid);
       if (entity != null) {
          entity.remove();
