--- net/minecraft/client/world/ClientWorld.java
+++ net/minecraft/client/world/ClientWorld.java
@@ -73,7 +73,7 @@
 import net.minecraft.world.storage.ISpawnWorldInfo;
 import net.minecraft.world.storage.MapData;
 
-public class ClientWorld extends World {
+public class ClientWorld extends World/* WDL >>> */ implements wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
    private final Int2ObjectMap<Entity> entitiesById = new Int2ObjectOpenHashMap<>();
    private final ClientPlayNetHandler connection;
    private final WorldRenderer worldRenderer;
@@ -116,6 +116,10 @@
       this.getProfiler().startSection("blocks");
       this.field_239129_E_.tick(hasTimeLeft);
       this.getProfiler().endSection();
+
+      /* WDL >>> */
+      wdl.WDLHooks.onWorldClientTick(this);
+      /* <<< WDL */
    }
 
    private void func_239141_x_() {
@@ -294,6 +298,10 @@
    }
 
    public void removeEntityFromWorld(int eid) {
+      /* WDL >>> */
+      wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, eid);
+      /* <<< WDL */
+
       Entity entity = this.entitiesById.remove(eid);
       if (entity != null) {
          entity.remove();
