--- net/minecraft/client/multiplayer/WorldClient.java
+++ net/minecraft/client/multiplayer/WorldClient.java
@@ -100,6 +100,10 @@
 		this.profiler.endStartSection("blocks");
 		this.tickBlocks();
 		this.profiler.endSection();
+
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientTick(this);
+		/* <<< WDL */
 	}
 
 	public void func_73031_a(int p_73031_1_, int p_73031_2_, int p_73031_3_, int p_73031_4_, int p_73031_5_, int p_73031_6_) {
@@ -167,6 +171,10 @@
 	}
 
 	public void doPreChunk(int chunkX, int chunkZ, boolean loadChunk) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientDoPreChunk(this, chunkX, chunkZ, loadChunk);
+		/* <<< WDL */
+
 		if (loadChunk) {
 			this.clientChunkProvider.loadChunk(chunkX, chunkZ);
 		} else {
@@ -249,6 +257,10 @@
 	}
 
 	public Entity removeEntityFromWorld(int entityID) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, entityID);
+		/* <<< WDL */
+
 		Entity entity = (Entity)this.entitiesById.removeObject(entityID);
 
 		if (entity != null) {
