--- net/minecraft/client/multiplayer/WorldClient.java
+++ net/minecraft/client/multiplayer/WorldClient.java
@@ -55,7 +55,7 @@
 import net.minecraft.world.storage.SaveHandlerMP;
 import net.minecraft.world.storage.WorldInfo;
 
-public class WorldClient extends World {
+public class WorldClient extends World/* WDL >>> */ implements wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
 	/** The packets that need to be sent to the server. */
 	private final NetHandlerPlayClient connection;
 	/** The ChunkProviderClient instance */
@@ -115,6 +115,10 @@
 		this.profiler.endStartSection("blocks");
 		this.tickBlocks();
 		this.profiler.endSection();
+
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientTick(this);
+		/* <<< WDL */
 	}
 
 	/**
@@ -262,6 +266,10 @@
 	}
 
 	public Entity removeEntityFromWorld(int entityID) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, entityID);
+		/* <<< WDL */
+
 		Entity entity = this.entitiesById.removeObject(entityID);
 
 		if (entity != null) {
