--- net/minecraft/client/multiplayer/WorldClient.java
+++ net/minecraft/client/multiplayer/WorldClient.java
@@ -1,6 +1,7 @@
 package net.minecraft.client.multiplayer;
 
 import com.google.common.collect.Sets;
+
 import java.util.Random;
 import java.util.Set;
 import javax.annotation.Nullable;
@@ -97,6 +98,10 @@
 		this.theProfiler.endStartSection("blocks");
 		this.updateBlocks();
 		this.theProfiler.endSection();
+
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientTick(this);
+		/* <<< WDL */
 	}
 
 	/**
@@ -168,6 +173,10 @@
 	}
 
 	public void doPreChunk(int chunkX, int chunkZ, boolean loadChunk) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientDoPreChunk(this, chunkX, chunkZ, loadChunk);
+		/* <<< WDL */
+
 		if (loadChunk) {
 			this.clientChunkProvider.loadChunk(chunkX, chunkZ);
 		} else {
@@ -251,6 +260,10 @@
 	}
 
 	public Entity removeEntityFromWorld(int entityID) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, entityID);
+		/* <<< WDL */
+
 		Entity entity = (Entity)this.entitiesById.removeObject(entityID);
 
 		if (entity != null) {
