--- net/minecraft/client/multiplayer/WorldClient.java
+++ net/minecraft/client/multiplayer/WorldClient.java
@@ -55,7 +55,7 @@
 import net.minecraft.world.storage.SaveHandlerMP;
 import net.minecraft.world.storage.WorldInfo;
 
-public class WorldClient extends World {
+public class WorldClient extends World/* WDL >>> */ implements wdl.ducks.IBaseChangesApplied/* <<< WDL */ {
    private final NetHandlerPlayClient connection;
    private ChunkProviderClient field_73033_b;
    private final Set<Entity> field_73032_d = Sets.newHashSet();
@@ -105,6 +105,10 @@
       this.profiler.endStartSection("blocks");
       this.func_147456_g();
       this.profiler.endSection();
+
+      /* WDL >>> */
+      wdl.WDLHooks.onWorldClientTick(this);
+      /* <<< WDL */
    }
 
    protected IChunkProvider createChunkProvider() {
@@ -226,6 +230,10 @@
    }
 
    public Entity removeEntityFromWorld(int p_73028_1_) {
+      /* WDL >>> */
+      wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, p_73028_1_);
+      /* <<< WDL */
+
       Entity entity = this.entitiesById.remove(p_73028_1_);
       if (entity != null) {
          this.field_73032_d.remove(entity);
