--- net/minecraft/inventory/InventoryBasic.java
+++ net/minecraft/inventory/InventoryBasic.java
@@ -9,7 +9,7 @@
 import net.minecraft.util.NonNullList;
 import net.minecraft.util.text.ITextComponent;
 
-public class InventoryBasic implements IInventory, IRecipeHelperPopulator {
+public class InventoryBasic implements IInventory, IRecipeHelperPopulator/* WDL >>> */, wdl.ducks.INetworkNameable/* <<< WDL */ {
    private final ITextComponent inventoryTitle;
    private final int slotsCount;
    private final NonNullList<ItemStack> inventoryContents;
@@ -19,11 +19,26 @@
     * instead
     */
    private ITextComponent customName;
+   /* WDL >>> */
+   @Nullable
+   private String networkCustomName;
 
+   @Nullable
+   @Override
+   public String getCustomDisplayName() {
+      return networkCustomName;
+   }
+   /* <<< WDL */
+
    public InventoryBasic(ITextComponent p_i45902_1_, int p_i45902_2_) {
       this.inventoryTitle = p_i45902_1_;
       this.slotsCount = p_i45902_2_;
       this.inventoryContents = NonNullList.withSize(p_i45902_2_, ItemStack.EMPTY);
+      /* WDL >>> */
+      if (p_i45902_1_ instanceof net.minecraft.util.text.TextComponentString) {
+         this.networkCustomName = p_i45902_1_.getString();
+      }
+      /* <<< WDL */
    }
 
    /**
