--- net/minecraft/block/BlockTNT.java
+++ net/minecraft/block/BlockTNT.java
@@ -20,6 +20,14 @@
 import net.minecraft.world.World;
 
 public class BlockTNT extends Block {
+	/* WDL - 1.13.1 >>> */
+	public static final net.minecraft.state.BooleanProperty UNSTABLE = net.minecraft.state.BooleanProperty.create("unstable");
+	@Override
+	protected void fillStateContainer(net.minecraft.state.StateContainer.Builder<Block, IBlockState> builder) {
+		builder.add(UNSTABLE);
+	}
+	/* <<< WDL */
+
 	public BlockTNT(Block.Builder builder) {
 		super(builder);
 	}
