--- net/minecraft/realms/RealmsSharedConstants.java
+++ net/minecraft/realms/RealmsSharedConstants.java
@@ -3,8 +3,12 @@
 import net.minecraft.util.SharedConstants;
 
 public class RealmsSharedConstants {
-	public static final int NETWORK_PROTOCOL_VERSION = 393;
+	/* WDL 1.13.1 >>> */
+	public static final int NETWORK_PROTOCOL_VERSION = 401;
+	/* <<< WDL */
 	public static final int TICKS_PER_SECOND = 20;
-	public static final String VERSION_STRING = "1.13";
+	/* WDL 1.13.1 >>> */
+	public static final String VERSION_STRING = "1.13.1";
+	/* <<< WDL */
 	public static final char[] ILLEGAL_FILE_CHARACTERS = SharedConstants.ILLEGAL_FILE_CHARACTERS;
 }
