--- net/minecraft/block/Block.java
+++ net/minecraft/block/Block.java
@@ -1686,6 +1686,13 @@
 		register("bubble_coral_block", new BlockCoral(block67, Block.Builder.create(Material.ROCK, MapColor.PURPLE).hardnessAndResistance(1.5F, 6.0F).sound(SoundType.CORAL)));
 		register("fire_coral_block", new BlockCoral(block68, Block.Builder.create(Material.ROCK, MapColor.RED).hardnessAndResistance(1.5F, 6.0F).sound(SoundType.CORAL)));
 		register("horn_coral_block", new BlockCoral(block69, Block.Builder.create(Material.ROCK, MapColor.YELLOW).hardnessAndResistance(1.5F, 6.0F).sound(SoundType.CORAL)));
+		/* WDL 1.13.1 >>> */
+		register("dead_tube_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.BLUE).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
+		register("dead_brain_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.PINK).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
+		register("dead_bubble_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.PURPLE).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
+		register("dead_fire_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.RED).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
+		register("dead_horn_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.YELLOW).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
+		/* <<< WDL */
 		register("tube_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.BLUE).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
 		register("brain_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.PINK).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
 		register("bubble_coral", new BlockCoralPlant(Block.Builder.create(Material.OCEAN_PLANT, MapColor.PURPLE).doesNotBlockMovement().zeroHardnessAndResistance().sound(SoundType.WET_GRASS)));
