--- net/minecraft/realms/RealmsScreen.java
+++ net/minecraft/realms/RealmsScreen.java
@@ -128,6 +128,14 @@
 		return this.proxy.height;
 	}
 
+	/* WDL 1.13.1 >>> */
+	public com.google.common.util.concurrent.ListenableFuture<Object> threadSafeSetScreen(RealmsScreen p_threadSafeSetScreen_1_) {
+		return this.minecraft.addScheduledTask(() -> {
+			Realms.setScreen(p_threadSafeSetScreen_1_);
+		});
+	}
+	/* <<< WDL */
+
 	public int fontLineHeight() {
 		return this.proxy.getFontHeight();
 	}
@@ -156,6 +164,12 @@
 		this.proxy.removeWidget(p_removeWidget_1_);
 	}
 
+	/* WDL 1.13.1 >>> */
+	public boolean hasWidget(RealmsGuiEventListener p_hasWidget_1_) {
+		return this.proxy.getChildren().contains(p_hasWidget_1_.getProxy());
+	}
+	/* <<< WDL */
+
 	public void buttonsAdd(RealmsButton p_buttonsAdd_1_) {
 		this.proxy.buttonsAdd(p_buttonsAdd_1_);
 	}
