--- net/minecraft/network/handshake/client/CPacketHandshake.java
+++ net/minecraft/network/handshake/client/CPacketHandshake.java
@@ -16,7 +16,9 @@
 	}
 
 	public CPacketHandshake(String p_i47613_1_, int p_i47613_2_, EnumConnectionState p_i47613_3_) {
-		this.protocolVersion = 393;
+		/* WDL 1.13.1 >>> */
+		this.protocolVersion = 401;
+		/* <<< WDL */
 		this.ip = p_i47613_1_;
 		this.port = p_i47613_2_;
 		this.requestedState = p_i47613_3_;
