--- net/minecraft/item/Item.java
+++ net/minecraft/item/Item.java
@@ -853,6 +853,13 @@
 		register(Blocks.BUBBLE_CORAL, ItemGroup.DECORATIONS);
 		register(Blocks.FIRE_CORAL, ItemGroup.DECORATIONS);
 		register(Blocks.HORN_CORAL, ItemGroup.DECORATIONS);
+		/* WDL 1.13.1 >>> */
+		register(Block.getBlockFromName("dead_brain_coral"), ItemGroup.DECORATIONS);
+		register(Block.getBlockFromName("dead_bubble_coral"), ItemGroup.DECORATIONS);
+		register(Block.getBlockFromName("dead_fire_coral"), ItemGroup.DECORATIONS);
+		register(Block.getBlockFromName("dead_horn_coral"), ItemGroup.DECORATIONS);
+		register(Block.getBlockFromName("dead_tube_coral"), ItemGroup.DECORATIONS);
+		/* <<< WDL */
 		register(new ItemWallOrFloor(Blocks.TUBE_CORAL_FAN, Blocks.TUBE_CORAL_WALL_FAN, (new Item.Builder()).group(ItemGroup.DECORATIONS)));
 		register(new ItemWallOrFloor(Blocks.BRAIN_CORAL_FAN, Blocks.BRAIN_CORAL_WALL_FAN, (new Item.Builder()).group(ItemGroup.DECORATIONS)));
 		register(new ItemWallOrFloor(Blocks.BUBBLE_CORAL_FAN, Blocks.BUBBLE_CORAL_WALL_FAN, (new Item.Builder()).group(ItemGroup.DECORATIONS)));
