--- net/minecraft/realms/Realms.java
+++ net/minecraft/realms/Realms.java
@@ -55,10 +55,27 @@
 		return Minecraft.getInstance().getSessionService().fillProfileProperties(new GameProfile(UUIDTypeAdapter.fromString(p_uuidToName_0_), (String)null), false).getName();
 	}
 
+	/* WDL 1.13.1 >>> */
+	public static <V> ListenableFuture<V> execute(java.util.concurrent.Callable<V> p_execute_0_) {
+		return Minecraft.getInstance().addScheduledTask(p_execute_0_);
+	}
+
+	public static void execute(Runnable p_execute_0_) {
+		Minecraft.getInstance().addScheduledTask(p_execute_0_);
+	}
+
 	public static void setScreen(RealmsScreen p_setScreen_0_) {
-		Minecraft.getInstance().displayGuiScreen(p_setScreen_0_.getProxy());
+		execute(() -> {
+			setScreenDirect(p_setScreen_0_);
+			return null;
+		});
 	}
 
+	public static void setScreenDirect(RealmsScreen p_setScreenDirect_0_) {
+		Minecraft.getInstance().displayGuiScreen(p_setScreenDirect_0_.getProxy());
+	}
+	/* <<< WDL */
+
 	public static String getGameDirectoryPath() {
 		return Minecraft.getInstance().gameDir.getAbsolutePath();
 	}
