--- net/minecraft/network/play/client/CPacketEditBook.java
+++ net/minecraft/network/play/client/CPacketEditBook.java
@@ -24,6 +24,9 @@
 	public void readPacketData(PacketBuffer buf) throws IOException {
 		this.field_210347_a = buf.readItemStack();
 		this.field_210348_b = buf.readBoolean();
+		/* WDL 1.13.1 >>> */
+		buf.readEnumValue(net.minecraft.util.EnumHand.class);
+		/* <<< WDL */
 	}
 
 	/**
@@ -32,6 +35,9 @@
 	public void writePacketData(PacketBuffer buf) throws IOException {
 		buf.writeItemStack(this.field_210347_a);
 		buf.writeBoolean(this.field_210348_b);
+		/* WDL 1.13.1 >>> */
+		buf.writeEnumValue(net.minecraft.util.EnumHand.MAIN_HAND);
+		/* <<< WDL */
 	}
 
 	/**
