--- net/minecraft/block/BlockConduit.java
+++ net/minecraft/block/BlockConduit.java
@@ -23,6 +23,16 @@
 import net.minecraft.world.World;
 
 public class BlockConduit extends BlockContainer {
+	/* WDL 1.13.1 >>> */
+	public static final net.minecraft.state.BooleanProperty WATERLOGGED = net.minecraft.state.properties.BlockStateProperties.WATERLOGGED;
+	@Override
+	protected void fillStateContainer(net.minecraft.state.StateContainer.Builder<Block, IBlockState> builder) {
+		builder.add(WATERLOGGED);
+	}
+	{
+		setDefaultState(getDefaultState().with(WATERLOGGED, true));
+	}
+	/* <<< WDL */
 	protected static final VoxelShape SHAPE = Block.makeCuboidShape(5.0D, 5.0D, 5.0D, 11.0D, 11.0D, 11.0D);
 
 	public BlockConduit(Block.Builder builder) {
