--- net/minecraft/client/multiplayer/WorldClient.java
+++ net/minecraft/client/multiplayer/WorldClient.java
@@ -102,6 +102,10 @@
 		this.profiler.endStartSection("blocks");
 		this.updateBlocks();
 		this.profiler.endSection();
+
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientTick(this);
+		/* <<< WDL */
 	}
 
 	/**
@@ -173,6 +177,10 @@
 	}
 
 	public void doPreChunk(int chunkX, int chunkZ, boolean loadChunk) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientDoPreChunk(this, chunkX, chunkZ, loadChunk);
+		/* <<< WDL */
+
 		if (loadChunk) {
 			this.clientChunkProvider.loadChunk(chunkX, chunkZ);
 		} else {
@@ -256,6 +264,10 @@
 	}
 
 	public Entity removeEntityFromWorld(int entityID) {
+		/* WDL >>> */
+		wdl.WDLHooks.onWorldClientRemoveEntityFromWorld(this, entityID);
+		/* <<< WDL */
+
 		Entity entity = this.entitiesById.removeObject(entityID);
 
 		if (entity != null) {
@@ -433,8 +445,8 @@
 		}
 	}
 
-	public void makeFireworks(double x, double y, double z, double motionX, double motionY, double motionZ, @Nullable NBTTagCompound compound) {
-		this.mc.effectRenderer.addEffect(new ParticleFirework.Starter(this, x, y, z, motionX, motionY, motionZ, this.mc.effectRenderer, compound));
+	public void makeFireworks(double x, double y, double z, double motionX, double motionY, double motionZ, @Nullable NBTTagCompound compund) {
+		this.mc.effectRenderer.addEffect(new ParticleFirework.Starter(this, x, y, z, motionX, motionY, motionZ, this.mc.effectRenderer, compund));
 	}
 
 	public void sendPacketToServer(Packet<?> packetIn) {
