diff -r -U 3 minecraft\net\minecraft\entity\passive\EquineEntity.java minecraft_patched\net\minecraft\entity\passive\EquineEntity.java
--- minecraft\net\minecraft\entity\passive\EquineEntity.java
+++ minecraft_patched\net\minecraft\entity\passive\EquineEntity.java
@@ -127,7 +127,7 @@
       this.field_70180_af.func_187214_a(field_184787_bE, Byte.valueOf((byte)0));
       this.field_70180_af.func_187214_a(field_184788_bF, Integer.valueOf(HorseType.HORSE.func_188595_k()));
       this.field_70180_af.func_187214_a(field_184789_bG, Integer.valueOf(0));
-      this.field_70180_af.func_187214_a(field_184790_bH, Optional.<T>absent());
+      this.field_70180_af.func_187214_a(field_184790_bH, Optional.<UUID>absent());
       this.field_70180_af.func_187214_a(field_184791_bI, Integer.valueOf(HorseArmorType.NONE.func_188579_a()));
    }
 
@@ -185,7 +185,7 @@
    }
 
    public void func_184779_b(@Nullable UUID p_184779_1_) {
-      this.field_70180_af.func_187227_b(field_184790_bH, Optional.<T>fromNullable(p_184779_1_));
+      this.field_70180_af.func_187227_b(field_184790_bH, Optional.fromNullable(p_184779_1_));
    }
 
    public float func_110254_bY() {
