diff -r -U 3 minecraft\net\minecraft\entity\passive\EquineEntity.java minecraft_patched\net\minecraft\entity\passive\EquineEntity.java
--- minecraft\net\minecraft\entity\passive\EquineEntity.java
+++ minecraft_patched\net\minecraft\entity\passive\EquineEntity.java
@@ -103,7 +103,7 @@
    protected void func_70088_a() {
       super.func_70088_a();
       this.field_70180_af.func_187214_a(field_184787_bE, Byte.valueOf((byte)0));
-      this.field_70180_af.func_187214_a(field_184790_bH, Optional.<T>absent());
+      this.field_70180_af.func_187214_a(field_184790_bH, Optional.<UUID>absent());
    }
 
    protected boolean func_110233_w(int p_110233_1_) {
@@ -130,7 +130,7 @@
    }
 
    public void func_184779_b(@Nullable UUID p_184779_1_) {
-      this.field_70180_af.func_187227_b(field_184790_bH, Optional.<T>fromNullable(p_184779_1_));
+      this.field_70180_af.func_187227_b(field_184790_bH, Optional.<UUID>fromNullable(p_184779_1_));
    }
 
    public float func_110254_bY() {
