diff -r -U 3 minecraft\net\minecraft\server\management\PlayerManager.java minecraft_patched\net\minecraft\server\management\PlayerManager.java
--- minecraft\net\minecraft\server\management\PlayerManager.java
+++ minecraft_patched\net\minecraft\server\management\PlayerManager.java
@@ -360,7 +360,7 @@
    }
 
    public class PlayerInstance {
-      private static final Logger field_187281_a = LogManager.getLogger();
+      private final Logger field_187281_a = LogManager.getLogger();
       private final List<EntityPlayerMP> field_187283_c = Lists.<EntityPlayerMP>newArrayList();
       private final ChunkPos field_187284_d;
       private final short[] field_187285_e = new short[64];
