--- net/minecraft/client/resources/DefaultResourcePack.java
+++ net/minecraft/client/resources/DefaultResourcePack.java
@@ -15,7 +15,7 @@
 import net.minecraft.util.ResourceLocation;
 
 public class DefaultResourcePack implements IResourcePack {
-	public static final Set<String> DEFAULT_RESOURCE_DOMAINS = ImmutableSet.<String>of("minecraft", "realms");
+	public static final Set<String> DEFAULT_RESOURCE_DOMAINS = ImmutableSet.<String>of("minecraft", "realms"/* WDL >>> */, "wdl" /* <<< WDL */);
 	private final ResourceIndex resourceIndex;
 
 	public DefaultResourcePack(ResourceIndex resourceIndexIn) {
